spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -160,6 +177,10 @@ BatchWrite asRewrite(String fileSetID) {
     return new RewriteFiles(fileSetID);
   }
 
+  BatchWrite asPositionDeletesRewrite(String fileSetId) {
```

Comment:
I don't think we share much of the logic in `SparkWriteBuilder` or in `SparkWrite`. What about creating separate hierarchy for that like we have for position deltas?

```
SparkPositionDeletesRewriteBuilder implements WriteBuilder ...
SparkPositionDeletesRewrite implements Write ...
```

Otherwise, we complicate this class even more.

Reply:
Refactored out